src/components/text-input/TextInput.tsx

```diff
@@ -42,6 +42,8 @@ const TextInput: React.FC<PTextInput> = ({
         animatedLabel,
         backgroundColor,
         width,
+        onClear,
```

Comment:
this should probably be named differently as well, since the trailingIcon will not always be a clear-icon, but instead can have a different action added to it.

The name is a bit clunky, but `handleTrailingIconClick` or `onTrailingIconClick` should be more descriptive of what it is supposed to do.

Reply:
This was another discussion I had with @michaelgamble that the trailing icon will always be a clearing one and would also need to appear only if there is a value applied to it. Maybe he can advice here too?

Reply:
this is kinda against our generic approach, but fair eonugh ... IMO this is just a small change with a potentially big impact! :D